modules/KIWILinuxRC.sh

```diff
@@ -881,6 +888,20 @@ function startPlymouth {
     fi
 }
 #======================================
+# startHaveged
+#--------------------------------------
+function startHaveged {
+    if [ -e /usr/sbin/haveged ]; then
+       /usr/sbin/haveged
+       return
+    fi
+
+    # a source of entropy is required for VMware virtual machines
+    if [ $fipsEnabled -eq 1 ]; then
+       echo "Warning: haveged is missing. The boot may hang if /dev/random is being read."
+    fi
+}
+#======================================
```

Comment:
Given the function is called only in case of fips enabled it could be imho written like the following

```bash
function startHaveged {
    if lookup haveged &>/dev/null; then
        if ! haveged; then
            systemException \
                "Failed to start haveged required for fips" \
            "reboot"
        fi
    else
        systemException \
            "haveged is missing but required for fips" \
        "reboot"
    fi
}
```

Thoughts ?